Sources/TuistKit/Cache/CacheController.swift

```diff
@@ -237,24 +237,34 @@ final class CacheController: CacheControlling {
 
         let graphTraverser = GraphTraverser(graph: graph)
 
-        return try topologicalSort(
+        let graph = try topologicalSort(
             Array(graphTraverser.allTargets()),
             successors: {
                 Array(graphTraverser.directTargetDependencies(path: $0.path, name: $0.target.name))
             }
         )
-        .filter { target in
-            guard let hash = hashesByCacheableTarget[target] else { return false }
-            let cacheExists = try cache.exists(hash: hash).toBlocking().first() ?? false
-            if cacheExists {
-                logger.pretty("The target \(.bold(.raw(target.target.name))) with hash \(.bold(.raw(hash))) and type \(cacheOutputType.description) is already in the cache. Skipping...")
+
+        let targetCacheExistsTuples = graph.map(context: .concurrent) { target -> (String, Bool)? in
```

Comment:
I would use a compactMap here, instead of doing it after
```suggestion
        let targetCacheExists: [String: Bool] = .init(uniqueKeysWithValues: graph.compactMap(context: .concurrent) { target -> (String, Bool)? in
```